### PR TITLE
Collect coverage from rq worker

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

File: pulpcore/tests/functional/worker_with_coverage.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import sys
```
Comment (mdellweg) [inline pulpcore/tests/functional/worker_with_coverage.py]: https://github.com/pulp/pulp/pull/3876#pullrequestreview-209132957

Not sure, i understand the request here. The two main benefits of having it i test are:
* It does not show up in production
* It does not show up in coverage

Comment (mdellweg) [reply pulpcore/tests/functional/worker_with_coverage.py]: ping to the new PR: @dralley @bmbouter

**Comment by bmbouter** [comment]
Thanks @mdellweg and @dralley !

**Comment by bmbouter** [comment]
This is still a big improvement since it picks up coverage in other areas of the worker. I believe we are still missing coverage from any worker task code that runs post RQ fork. For instance I always check the _queue_reserve_resource code which runs with every Pulp task. https://codecov.io/gh/pulp/pulpcore/pull/10/src/pulpcore/tasking/tasks.py#L77

I'm pointing it out just to share the info, I don't expect anyone to jump in and resolve. Speaking of which I'm not sure how to resolve it. It would probably be good to open a bug against the RQ community about their recommendation on getting coverage info from RQ tasks.

**Comment by bmbouter** [comment]
I filed an issue here if you want to watch it:  https://github.com/rq/rq/issues/1051

**Comment by bmbouter** [comment]
@pulp/qe FYI

**Comment by daviddavis** [comment]
This change has doubled the time it takes for Travis builds from 20 min to 40 min. I'd propose we revert this change temporarily until we can figure out why that is. I should have some time after the RC release to investigate. Here's the PR:

https://github.com/pulp/pulpcore/pull/19

**Pull request merged**